File: app/src/main/java/com/montywu/demo/componentsbasicsample/ui/main/MainViewModel.java
package com.montywu.demo.componentsbasicsample.ui.main;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;

import com.montywu.demo.componentsbasicsample.entity.User;

/**
 * @author monty
 */
public class MainViewModel extends ViewModel {
    private MutableLiveData<User> user;

    public MutableLiveData<User> getUser() {
        if (user == null) {
            user = new MutableLiveData<>();
        }
        return user;
    }
}

File: livedata/src/main/java/com/montywu/demo/livedata/ui/main/MainViewModel.kt
package com.montywu.demo.livedata.ui.main

import android.arch.lifecycle.MutableLiveData
import android.arch.lifecycle.ViewModel

class MainViewModel : ViewModel() {
    private val _data = MutableLiveData<String>()

    val data:MutableLiveData<String>
        get() = _data

    init {
        _data.value = "Hello Monty!"
    }
}
